module/setter.go

```diff
@@ -70,11 +62,19 @@ if !dstOk && dst.%s != nil {
 			fPath = m.ctx.Name(f.OneOf()).String()
 		}
 
-		buildIndented(buf, tabCount, fmt.Sprintf(`if src != nil {
+		if f.InOneOf() {
```

Comment:
We need to check if the source type actually matches, not only that it is present.